===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_asn1_defs_load:defs->buckets)) at 21 (line 81, column 7),
        Insert: if (true) free(*(p11_asn1_defs_load:defs)) at 23 (line 84, column 4) }
===== Actual patch in diff format =====
@@ -79,8 +79,10 @@ p11_asn1_defs_load (void)
 		def = NULL;
 		ret = asn1_array2tree (asn1_tabs[i].tab, &def, message);
 		if (ret != ASN1_SUCCESS) {
+			free(defs->buckets);
 			p11_debug_precond ("failed to load %s* definitions: %s: %s\n",
 			                   asn1_tabs[i].prefix, asn1_strerror (ret), message);
+			free(defs);
 			return NULL;
 		}
 
